packages/taquito-michelson-encoder/src/tokens/token.ts

```diff
@@ -1,6 +1,10 @@
 import { MichelsonV1Expression } from '@taquito/rpc';
 import { TokenSchema } from '../schema/types';
 
+/**
+ *  @category Error
+ *  @description Error that indicates an invalid token being passed or used
```

Comment:
I don't think that the users are necessarily familiar with the term token; tokens are created by the Michelson Encoder to represent Michelson types and are used internally. We can add precision on the description.  We throw a TokenValidationError when the MichelsonEncoder fails to encode invalid/incorrect data. Encoding here means transforming js data into Michelson JSON format. For example, if an address is expected but a number is received, then we throw a ValidationError.

Reply:
that's a good point. I can update it to have a friendlier description